### PR TITLE
test: extend When_CreatingBatch_Then_BatchIsEventuallyCompleted

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
This PR would extend the verification of the flow of creating a batch to include the zip and download steps.

## References

* #0000

### CODE REVIEW

File: source/dotnet/Services/DomainTests/WebApiTests.cs [hunk 1]
```diff
@@ -132,11 +134,31 @@ public async Task When_CreatingBatch_Then_BatchIsEventuallyCompleted()
...
+
+                result.Should().NotContainNullsOrEmptyEnumerables();
+
+                var basisData = await Fixture.WholesaleClient.GetZippedBasisDataStreamAsync(batchId)
+                    .ConfigureAwait(false);
+                var archive = new ZipArchive(basisData);
+
+                var basisDataFileNames = archive.Entries.Select(entry => entry.FullName).ToList();
+
+                basisDataFileNames.Should().Contain("Timeseries_PT1H.csv")
```
Comment (BjarkeMeier) [inline source/dotnet/Services/DomainTests/WebApiTests.cs:158]: What about `result.json`?

File: source/dotnet/Services/DomainTests/WebApiTests.cs [hunk 2]
```diff
@@ -132,11 +134,31 @@ public async Task When_CreatingBatch_Then_BatchIsEventuallyCompleted()
...
+                    .GetProcessStepResultAsync(
+                        new ProcessStepResultRequestDto(
+                            batchId,
+                            ExistingGridAreaCode,
+                            ProcessStepType.AggregateProductionPerGridArea))
+                    .ConfigureAwait(false);
+
+                result.Should().NotContainNullsOrEmptyEnumerables();
+
+                var basisData = await Fixture.WholesaleClient.GetZippedBasisDataStreamAsync(batchId)
```
Comment (BjarkeMeier) [inline source/dotnet/Services/DomainTests/WebApiTests.cs:152]: The test has been greatly extended but the test name hasn't changed.
Do you think the test name is still correct/suitable?

**Review by BjarkeMeier — COMMENTED** [review]
Could you extend the PR description regarding _why_ you want this testing?

File: source/dotnet/Services/DomainTests/WebApiTests.cs (hunk 1 above)
Comment (lasrinnil) [reply source/dotnet/Services/DomainTests/WebApiTests.cs:158]: the result is not a part of the basis data, it has its own client call.

File: source/dotnet/Services/DomainTests/WebApiTests.cs (hunk 2 above)
Comment (lasrinnil) [reply source/dotnet/Services/DomainTests/WebApiTests.cs:152]: I will try to update the name accordingly :)

File: source/dotnet/Services/DomainTests/WebApiTests.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/Services/DomainTests/WebApiTests.cs:158]: But it's still in the zip

**Pull request closed without merging**